src/services/services.ts

```diff
@@ -2808,6 +2811,22 @@ namespace ts {
                     }
                 }
 
+                // Special case for function expression and class expression because despite sometimes having a name, the binder
+                // binds them to a symbol with the name "__function" and "__class" respectively. However, for completion entry, we want
+                // to display its declared name rather than "__function" and "__class".
+                //      var x = function foo () {
+                //          fo$  <- completion list should contain local name "foo"
+                //      }
+                //      foo$ <- completion list should not contain "foo"
+                if (displayName === "__function" || displayName === "__class") {

```

Comment:
```getDeclaredName``` doesn't do the right thing for functionExpression. Talk with @mhegazy, he says there will be some rewrite on that. So I don't want to have all the change in this PR